test/jdk/tools/jpackage/share/AppLauncherEnvTest.java

```diff
@@ -83,8 +83,14 @@ public static void test() throws Exception {
         final String expectedEnvVarValue = Optional.ofNullable(System.getenv(
                 envVarName)).orElse("") + File.pathSeparator + appDir;
 
-        TKit.assertEquals(expectedEnvVarValue, actualEnvVarValue, String.format(
-                "Check value of %s env variable", envVarName));
+        String msg = String.format("Check value of %s env variable", envVarName);
+        if (TKit.isLinux()) {
+            if (! actualEnvVarValue.endsWith(expectedEnvVarValue)) {
+                TKit.assertTrue(false, msg);
+            }
+        } else {
+            TKit.assertEquals(expectedEnvVarValue, actualEnvVarValue, msg);
+        }
```

Comment:
I'd keep the check as a single statement:
```
TKit.assertTextStream(expectedEnvVarValue)
    .predicate(TKit.isLinux() ? String::endsWith : String::equals)
    .label(String.format("value of %s env variable", envVarName))
    .apply(Stream.of(actualEnvVarValue));
```

It also will produce a nicer log record than `TKit.assertTrue(false, msg);`

Reply:
Hi Alexey, I adjusted the check following your advice .